### DISCUSSION TITLE
Dew Heater not working?

### DISCUSSION THREAD

**Discussion by Pedritoprint** [AllskyTeam/allsky-modules#139] [General]
I installed the "User Modules" as specified [here](https://github.com/AllskyTeam/allsky-modules) but the installer  (_./install.sh_) sends an ERROR **-bash: ./install.sh: file or directory not found**. Then i changed into _**/allsky-modules**_ and made the installation from there but it seems, that, after installation of the "Dew Heater-Module", it isn't running. I have no Temp and Humity on the programmed overlay, only "???"
I2C Adress 0x77 is ok.
Btw, the requirements are also installed. (BME280, meteocalc) 
My System Raspberry Pi 4 4Gb, OS Bookworm and your latest Allsky-SW.
**Edit:** Debuginfo: failed to read sensor
**Edit:** If i start manually "allsky-dewheater.py then --> **import allsky_shared as s**, **ModuleNotFoundError: No module named 'allsky_shared'**
**Edit:** 2.6.24....sorry, i think i have found this morning the reason above, i tested an older BM**P**280 instead of a BME280 i ordered. Just waiting of it 
                           sorry sorry sorry
**Edit** 7.6.2024: I just now tested a new BME280 but, as writen above, The same behavior. Adress is 0x76 (changed in Modul.Settings). Debuginfo: failed to read Sensor.

**Comment by Alex-developer** [comment]
Can you please let me know the version of Allsky you are running and also set the debug level to 4 in the settings and then paste the error codes from the /var/log/allsky.log file

Thanks

**Comment by Pedritoprint** [comment]
Allsky-Version: v2023.05.01_04
[allsky.log](https://github.com/user-attachments/files/15745069/allsky.log)

Edit: Today i tested the BME280 with a fresh installed Raspi OS-Bookworm and i think, the problem is the newest OS, because i can't use several Test-routines for the BME280 i found. 
It's possible, that the "virtual environment" is the problem. I also have, during installation of Adafruit-Modul, the "_error :externally-managed-environment_" wich i have bypassed with **"sudo rm /usr/lib/python3.11/EXTERNALLY-MANAGED"**
The BME280 is mounted and seen on address 0x76. Also the read out with  "_i2cdump -y -f 1 0x76_" displays some informations.

**Comment by Alex-developer** [comment]
Allsky and the modules need a virtual environment. 

Do you have a 'venv' folder in the ~/allsky folder?
If you do you could re install the modules which will ensure that everything the modules require is correctly installed

**Reply by Pedritoprint** [reply]
I think, i had the "venv"-Folder but now, i have to do a new setup for my Raspi in case of a mistake myself. I deleted the SD-Card, so i'm doing just now a new setup. I will inform you asap if i have the new installation running.
**Edit:** I have now reinstalled a freshly new allsky with allsky-modules. The cam is working fine, website also, but the same issue with the dewheater-modul.
I see the **/allsky/venv/** -folder with a lot of files and folders inside.
BTW, i think, that the install-commands for the modules [here](https://github.com/AllskyTeam/allsky-modules) have missing a **cd /allsky-modules/** -command before **_./install.sh_**, because there is always an error for the missing file. 
I have the same Raspi and the same config like described above and now the same "**failed to read sensor** in the **module Manager's debug info**".
Just now, i installed the GPS-Module and it seems, that this modul is working fine.
greetings, Peter
Raspi: 4B 4Gb Rev 1.5
OS: Bookworm
Allsky Vers: v2023.05.01_04

**Reply by Alex-developer** [reply]
This is very strange. Following your re install of Allsky is there anything logged in the allskyperiodic log file related to the dew heater module?

Regarding the missing cd when you say there is an 'error for the missing file' what error is displayed?

Thanks

Alex

**Reply by Pedritoprint** [reply]
I gave up further tests with my cheap chinese BME280 so i have ordered new ones  (original Versions of BME280, coming today), but following Error is in allskyperiodic log: 
```
INFO: Attempting to load allsky_dewheater
Jun 16 17:48:55 allsky-mob allskperiodic[11575]: ERROR: Module readBme280I2C failed on line 615 - No I2C device at address: 0x76
Jun 16 17:48:55 allsky-mob allskperiodic[11575]: ERROR: Failed to read sensor
Jun 16 17:48:55 allsky-mob allskperiodic[11575]: INFO: Module allsky_dewheater.py ran ok in 0.00s
```
but if i test the I2C with `i2cdetect -y 1` the device is displaed at adr 0x76
Additionaly i tested the actual BME280 i have with Raspberry Pi4 and OS Bullseye, standalone without allsky installed, only with test routines i found on the net, without success. So my conclusion: this BME280 is not like a BME280

_**Regarding the missing cd when you say there is an 'error for the missing file' what error is displayed?**_ 
After the `git clone https://github.com/AllskyTeam/allsky-modules.git` and then
` ./install.sh` like described [here](https://github.com/AllskyTeam/allsky-modules)
after the git-command i have to change the directory with `cd /allsky-modules` and then run the installer.
Peter

**Reply by Pedritoprint** [reply]
Dear Alex, i don't know how to excuse me about this discussion i started. As i guessed above, it was the cheap chinese Sensor. Today i received a other BME280, originall Adafruit and what surprise, the Sensor now is readable and working in OS Bookworm.
Anyway, thanks for your effort.
I close this (my) discussion now.

**Reply by Alex-developer** [reply]
> Dear Alex, i don't know how to excuse me about this discussion i started. As i guessed above, it was the cheap chinese Sensor. Today i received a other BME280, originall Adafruit and what surprise, the Sensor now is readable and working in OS Bookworm. Anyway, thanks for your effort. I close this (my) discussion now.

Glad you have it working :-)

I will update the documentation as you are correct in that you have to change into the allsky-modules directory before running the installer
